src/routes/index.svelte

```diff
@@ -37,12 +37,47 @@
 	async function updateEntries() {
 		fetchIndex += 1
 		let thisFetchIndex = fetchIndex
-		const res = await fetch(
-			`/api/entries.json?visible=${showVisible}&unlisted=${showUnlisted}&hidden=${showHidden}`
-		)
+		let search_value = (document.getElementById('search') as HTMLInputElement).value || null
+
+		let tags: string
+		let query: string
+		if (!search_value) {
+			query = undefined
+		} else {
+			let all = search_value.split(' ')
+			query = all.filter(word => !word.startsWith('tags:')).join(' ')
+			if (search_value.includes('tags:')) {
+				let tags_raw = all.filter(word => word.startsWith('tags:'))[0]
+				tags = tags_raw.slice(5).replaceAll('_', '%20')
+				console.log(query)
+			}
+		}
+
+		let url = `/api/entries.json?visible=${showVisible}&unlisted=${showUnlisted}&hidden=${showHidden}`
+
+		if (query) {
+			url += `&query=${query}`
+		} else if (tags) {
+			url += `&tags=${tags}`
+		}
+
+		const res = await fetch(url)
 		const newEntries = await res.json()
 
-		if (thisFetchIndex === fetchIndex) entries = newEntries
+		if (newEntries.length === 0) {
+			entries = [
+				{
+					title: 'No Results Found',
+					content: 'Sorry, there are no results found with that query!',
+					slug: 'entry_does_not_exist',
+					id: '0',
+					tags: ['Error'],
+					visibility: 'visible',
+				},
+			]
+		} else {
+			if (thisFetchIndex === fetchIndex) entries = newEntries
+		}
```

Comment:
I don't really like how this makes a fake entry, I think it'd be better if an {:else} was added to the `{#each entries as entry} <EntryPreview {entry} /> {/each}` that showed the error message.

Reply:
I agree, and yes. I can make this actually show a error message, will do in next commit. If possible